### PR TITLE
Sampling window, standard error for metrics, and SNR

### PR DESCRIPTION
**Pull request by yahskapar**
Pretty straightforward features that I tried to test thoroughly. I encourage everyone to carefully take a look at the logic in place (for both sampling windows and the changes to metrics) in case I missed some kind of edge case. The SNR metric implementation is based on a frequent equation for SNR that I saw in a few papers, chiefly this one on the bottom of page 6: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7899506/pdf/boe-12-1-494.pdf. The standard error is a very standard formulation for standard error that is simply:

$$standard error = \frac{\sigma}{\sqrt{n}}$$

Where $\sigma$ is the standard deviation and $n$ is the number of samples which could simply be the number of videos tested on or, if a smaller sampling window is used, a larger number based on how a video is chunked based on the window size at evaluation time. For the Pearson metric, a different formulation is used based on this thread: https://stats.stackexchange.com/questions/226380/derivation-of-the-standard-error-for-pearsons-correlation-coefficient.

Some results from testing the SNR and standard errors:

UBFC tested on PURE:
FFT MAE (FFT Label): 3.694385593220339 +/- 1.7376276602140517
FFT RMSE (FFT Label): 13.848831292145585 +/- 113.83706892303275
FFT MAPE (FFT Label): 3.3882176381266618 +/- 1.4376647949313572
FFT Pearson (FFT Label): 0.8176110623563922 +/- 0.07626272119507735
FFT SNR (FFT Label): 5.261351780678735 +/- 1.1099221524725582

UBFC After motion augmentation tested on PURE:
FFT MAE (FFT Label): 1.1172537076271187 +/- 0.7542595904048776
FFT RMSE (FFT Label): 5.900322033631117 +/- 33.75286796464523
FFT MAPE (FFT Label): 1.2770134322717948 +/- 0.8337279906070219
FFT Pearson (FFT Label): 0.9689888394596143 +/- 0.03272978926468511
FFT SNR (FFT Label): 8.942770905908834 +/- 0.938324945843469

Some results from testing the sampling window:

UBFC after motion augmentation tested on PURE (No smaller sampling window) [Run-time: 1 minute and 14 seconds]:
FFT MAE (FFT Label): 1.1172537076271187 +/- 0.7542595904048776
FFT RMSE (FFT Label): 5.900322033631117 +/- 33.75286796464523
FFT MAPE (FFT Label): 1.2770134322717948 +/- 0.8337279906070219
FFT Pearson (FFT Label): 0.9689888394596143 +/- 0.03272978926468511
FFT SNR (FFT Label): 8.942770905908834 +/- 0.938324945843469

UBFC after motion augmentation tested on PURE (window size of 10 seconds) [Run-time: 8minutes, 4 seconds ]:
FFT MAE (FFT Label): 1.0167738970588236 +/- 0.26791616677265073
FFT RMSE (FFT Label): 5.506331873792725 +/- 11.969440834175716
FFT MAPE (FFT Label): 1.1608005152069096 +/- 0.2961639550569115
FFT Pearson (FFT Label): 0.9725642349548783 +/- 0.011545456198612055
FFT SNR (FFT Label): 8.777935777910466 +/- 0.35833530490441406

UBFC after motion augmentation tested on PURE (window size of 30 seconds) [Run-time: 2 minutes, 59 seconds]:
FFT MAE (FFT Label): 0.9603942466887417 +/- 0.4189477968348463
FFT RMSE (FFT Label): 5.236932578949375 +/- 18.68587654474834
FFT MAPE (FFT Label): 1.093858267304527 +/- 0.4631281791815995
FFT Pearson (FFT Label): 0.9750717940066517 +/- 0.01817790467407979
FFT SNR (FFT Label): 8.700640980570325 +/- 0.5940321031599473

### CODE REVIEW

File: unsupervised_methods/unsupervised_predictor.py [hunk 1]
```diff
@@ -45,54 +46,94 @@ def unsupervised_predict(config, data_loader, method_name):
...
+                if window_frame_size > video_frame_size:
+                    window_frame_size = video_frame_size
+            else:
+                window_frame_size = video_frame_size
+
+            for i in range(0, len(BVP), window_frame_size):
+                BVP_window = BVP[i:i+window_frame_size]
+                label_window = labels_input[i:i+window_frame_size]
+
+                if len(BVP_window) < 9:
```
Comment (yahskapar) [inline unsupervised_methods/unsupervised_predictor.py:61]: This is definitely a bit hacky, but basically using smaller windows if there's ever a case where the final window (which could have a maximum size of 10 seconds, or 300 frames if the video has frame-rate of 30 Hz) is only filled to 9 values or fewer, scipy's `filtfilt` function throws an error related to the padding length needing to be greater than 9. I have no idea where the number 9 comes from, even after some investigation into the usage of `filtfilt` and similar errors that came up in search results.

Instead of potentially ignoring 9 or fewer frames for any given video, we could also just pad with 0 or somehow extend the signal (interpolate?) based on the previous window if needed. Curious if there are any strong thoughts on this.

**Review by xliucs — APPROVED** [review]
LGTM
